src/Command/BaseCommand.php

```diff
@@ -13,78 +13,47 @@
 
 namespace Sonata\PageBundle\Command;
 
+use Psr\Container\ContainerInterface;
 use Sonata\Doctrine\Model\ManagerInterface;
 use Sonata\NotificationBundle\Backend\BackendInterface;
-use Sonata\PageBundle\CmsManager\CmsPageManager;
-use Sonata\PageBundle\CmsManager\DecoratorStrategyInterface;
+use Sonata\PageBundle\CmsManager\CmsManagerInterface;
 use Sonata\PageBundle\Listener\ExceptionListener;
 use Sonata\PageBundle\Model\PageManagerInterface;
 use Sonata\PageBundle\Model\SiteManagerInterface;
 use Sonata\PageBundle\Model\SnapshotManagerInterface;
-use Symfony\Bundle\FrameworkBundle\Command\ContainerAwareCommand;
+use Symfony\Component\Console\Command\Command;
 use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Contracts\Service\ServiceSubscriberInterface;
 
 /**
  * BaseCommand.
  *
  * @author Thomas Rabaix <thomas.rabaix@sonata-project.org>
  */
-abstract class BaseCommand extends ContainerAwareCommand
+abstract class BaseCommand extends Command implements ServiceSubscriberInterface
 {
-    /**
-     * @return SiteManagerInterface
-     */
-    public function getSiteManager()
-    {
-        return $this->getContainer()->get('sonata.page.manager.site');
-    }
+    /** @var ContainerInterface */
+    protected $locator;
 
-    /**
-     * @return PageManagerInterface
-     */
-    public function getPageManager()
+    public function __construct(ContainerInterface $locator)
```

Comment:
when need to avoid to inject the whole ContainerInterface, as far as I know it will be removed for the future symfony releases, and I guess to this work the service need to be public!

Reply:
hmmm interesting, I didn't know about it!
well I need to research a bit more about this, But I would inject all dependencies explicit in my service!

But I do not have any argument to say that it's right or wrong, I just like to look at the class and see what is necessary for this work!

Reply:
I have suggested to use Service Subscriber because there is set of services here and commands use only some from the set. The benefit is that only used services are initialized not all from the set. Commands should be also lazy and private services.